packages/astro/src/content/loaders/glob.ts

```diff
@@ -63,41 +64,75 @@ export function glob(globOptions: GlobOptions): Loader {
 
 	const fileToIdMap = new Map<string, string>();
 
-	async function syncData(
-		entry: string,
-		base: URL,
-		{ logger, parseData, store }: LoaderContext,
-		entryType?: ContentEntryType
-	) {
-		if (!entryType) {
-			logger.warn(`No entry type found for ${entry}`);
-			return;
-		}
-		const fileUrl = new URL(entry, base);
-		const { body, data } = await entryType.getEntryInfo({
-			contents: await fs.readFile(fileUrl, 'utf-8'),
-			fileUrl,
-		});
-
-		const id = generateId({ entry, base, data });
-
-		const filePath = fileURLToPath(fileUrl);
-
-		const parsedData = await parseData({
-			id,
-			data,
-			filePath,
-		});
-
-		store.set(id, parsedData, body, filePath);
-
-		fileToIdMap.set(filePath, id);
-	}
-
 	return {
 		name: 'glob-loader',
 		load: async (options) => {
-			const { settings, logger, watcher } = options;
+			const { settings, logger, watcher, parseData, store } = options;
+			const { h64ToString } = await xxhash();
+
+			const renderFunctionByContentType = new WeakMap<ContentEntryType, RenderFunction>();
+
+			const untouchedEntries = new Set(options.store.keys());
+
+			async function syncData(entry: string, base: URL, entryType?: ContentEntryType) {
+				if (!entryType) {
+					logger.warn(`No entry type found for ${entry}`);
+					return;
+				}
+				const fileUrl = new URL(entry, base);
+				const contents = await fs.readFile(fileUrl, 'utf-8');
+
+				const { body, data } = await entryType.getEntryInfo({
+					contents,
+					fileUrl,
+				});
+
+				const digest = h64ToString(contents);
+
+				const id = generateId({ entry, base, data });
+				untouchedEntries.delete(id);
+
+				const existingEntry = store.get(id);
+
+				if (existingEntry && existingEntry.digest === digest) {
+					return;
+				}
+
+				const filePath = fileURLToPath(fileUrl);
+
+				const parsedData = await parseData({
+					id,
+					data,
+					filePath,
+				});
+
+				if (entryType.getRenderFunction) {
+					let render = renderFunctionByContentType.get(entryType);
```

Comment:
It seems we have two ways to access the render function:
- `renderFunctionByContentType.get(entryType)`
- `entryType.getRenderFunction(settings)`

What was the motivation to have both?

Reply:
I am not sure I understand the implications of this choice. Can we add some tech documentation somewhere that explains when to use one and when to use the other?

Reply:
To be clear: this isn't two different APIs. The API is `entryType.getRenderFunction(settings)`, and I'm just memoising it in a Map.